platform/ecs/agent/client.go

```diff
@@ -90,6 +91,11 @@ func (c *client) newRequest(endpoint string) (*http.Request, error) {
 }
 
 func decodeBody(resp *http.Response, out interface{}) error {
-	defer resp.Body.Close()
+	if resp.Body != nil {
+		defer resp.Body.Close()
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("incorrect status code %d", resp.StatusCode)
```

Comment:
It reminds me incorrect status code is a code like 600, so I would use just `fmt.Errorf("got %s", resp.Status)`.